crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -568,3 +568,99 @@ pub fn resolve_assignment<'a>(
         _ => None,
     }
 }
+
+/// Get the assigned [`Expr`] value of a given id, if any.
+///
+/// For example given
+/// ```python
+///  foo = 42
+///  (bar, bla) = 1, "str"
+/// ```
+///
+/// This function will return a `NumberLiteral` with value `Int(42)` when called with `foo` and a
+/// `StringLiteral` with value `"str"` when called with `bla`.
+pub fn get_assigned_value<'a>(id: &str, semantic: &'a SemanticModel<'a>) -> Option<&'a Expr> {
+    let scope = semantic.current_scope();
+    let binding_id = scope.get(id)?;
+    let binding = semantic.binding(binding_id);
+    if binding.kind.is_assignment() || binding.kind.is_named_expr_assignment() {
+        let parent_id = binding.source?;
+        let parent = semantic.statement(parent_id);
+        match parent {
+            Stmt::Assign(ast::StmtAssign { value, targets, .. }) => match value.as_ref() {
+                Expr::Tuple(ast::ExprTuple { elts, .. })
+                | Expr::List(ast::ExprList { elts, .. })
+                | Expr::Set(ast::ExprSet { elts, .. }) => {
+                    if let Some(target) = targets.iter().next() {
```

Comment:
What happens here in the case of multi-target assignments?

Reply:
Good catch. It works for 

```python
x = y = 0
```

Added example of that to TRIO115 fixture

Does not work for 

```python
(x, y) = [a, b] = (0, 0)
```

Which will only manage to retrieve the values for `x` and `y` right now. 

Would it be an idea to set up unit tests for this and the other new func in the module? If so, is there a leading example of non plugin unit tests I can have a look at?

Reply:
Is it able to detect `y` in `x = y = 0`?

Reply:
Yeah, we don't have great infrastructure for this. I'd probably just recommend adding more trio test cases as a means of testing this behavior.

Reply:
> Is it able to detect `y` in `x = y = 0`?

Yeah this one works. But if one of the targets is a collection there are issues. e.g.

```python
a = (x, y) = (0, 0)
(a, b) = [c, d] = [0, 0]
```
Not entirely sure how to best fix that. Flag it as a multi assign if the top-level targets has a length larger than 1 and add a separate branch for that? Or are there single target scenarios that could be mistaken then?